### PR TITLE
Implement dereferencing using Eidolon

### PR DESCRIPTION
**Pull request by danielgtaylor**
This change updates Attributes Kit with an optional parameter (`options.dataStructures`) to use [Eidolon](https://github.com/danielgtaylor/eidolon) to dereference/normalize the refract element being rendered.

The playground has been updated to add a toggle for turning this feature on and off. Other than that, no UI changes have been made (e.g. we do not show where the referenced data is coming from yet).

screen shot 2016-02-04 at 14 28 25

cc @Baggz

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -115,5 +115,8 @@
     "svgo": "0.6.1",
     "webpack": "1.12.12",
     "webpack-dev-server": "1.14.1"
+  },
+  "dependencies": {
+    "eidolon": "^1.2.0"
```
Comment (XVincentX) [inline package.json]: This should be in `devDependencies`

File: playground/components/Playground.js [hunk 2]
```diff
@@ -46,13 +58,18 @@ class Playground extends React.Component {
...
         <div className="column">
           <JsonFormatterComponent
-            element={this.state.parseResult.attributes}
+            element={this.state.parseResult.dataStructures[0]}
+            dataStructures={dataStructures}
           />
         </div>
 
         <div className="column">
+          <div style={{paddingBottom: '4px', marginBottom: '4px', borderBottom: '1px solid #ddd'}}>
```
Comment (XVincentX) [inline playground/components/Playground.js]: I am not sure we should style elements in this way.

File: src/index.js [hunk 3]
```diff
@@ -54,7 +54,7 @@ class AttributesKit extends EventEmitter {
   }
 
   render(refractElement) {
-    ReactDom.render(<AttributesComponent element={refractElement} />, this.element);
+    ReactDom.render(<AttributesComponent element={refractElement} dataStructures={this.options.dataStructures} />, this.element);
```
Comment (XVincentX) [inline src/index.js]: Is this an interface change?

**Comment by XVincentX** [comment]
Very cool! This is something that will really help to move forward.

File: package.json (hunk 1 above)
Comment (danielgtaylor) [inline package.json]: Nice catch. Does this mean Attributes Kit can never be `npm install`ed and used server-side in Node?

File: playground/components/Playground.js (hunk 2 above)
Comment (danielgtaylor) [inline playground/components/Playground.js]: Hmm I'll update this.

File: src/index.js (hunk 3 above)
Comment (danielgtaylor) [inline src/index.js]: Yes, as described in the PR description there is a new `dataStructures` option that, if set, will be used to dereference/resolve/normalize the refract structure. I attempted to make it backward-compatible, so it is optional.

File: package.json (hunk 1 above)
Comment (XVincentX) [inline package.json]: Yes you can, using the [server build](https://github.com/apiaryio/attributes-kit/blob/master/package.json#L34)

**Comment by Baggz** [comment]
@danielgtaylor 

Looks good to me! I'll try to integrate this branch into Apiary and see how it goes.

File: src/index.js (hunk 3 above)
Comment (Baggz) [inline src/index.js]: @XVincentX 

Just a note—this change is not breaking the existing API.

**Pull request merged**